python-package/xgboost/spark/core.py

```diff
@@ -262,6 +260,13 @@ def _validate_params(self):
                 "Therefore, that parameter will be ignored."
             )
 
+        if self.getOrDefault(self.features_cols) and not self.getOrDefault(
+            self.use_gpu
+        ):
+            raise ValueError(
+                "XGBoost accepts a list of feature column names only when use_gpu is enabled."
+            )
```

Comment:
We should emit warning say if "features_cols" param set, then "features_col" param is igored.

Reply:
Done